### PR TITLE
Edit branch protection

### PR DESCRIPTION
**Pull request by bjoernhaeuser**
This adds support for editing branch protection.

I also polished the integration tests and added a new public key (which
now has the required length, 2048 bits).

This also adds the integration tests for branch protection.

### CODE REVIEW

**Comment by bjoernhaeuser** [comment]
Is there anything still open? I would love to merge this :-)

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ go-github is a Go client library for accessing the [GitHub API][].
 **Build Status:** [![Build Status](https://travis-ci.org/google/go-github.svg?branch=master)](https://travis-ci.org/google/go-github)  
 **Test Coverage:** [![Test Coverage](https://coveralls.io/repos/google/go-github/badge.svg?branch=master)](https://coveralls.io/r/google/go-github?branch=master) ([gocov report](https://drone.io/github.com/google/go-github/files/coverage.html))
 
-go-github requires Go version 1.1 or greater.
+go-github requires Go version 1.4 or greater.
```
Comment (parkr) [inline README.md]: This seems unrelated to your change.

File: tests/integration/users_test.go [hunk 2]
```diff
@@ -173,8 +173,8 @@ func TestUsers_Keys(t *testing.T) {
 		t.Fatalf("Users.ListKeys('') returned error: %v", err)
 	}
 
-	// ssh public key for testing (fingerprint: 04:11:3a:3d:65:ed:f5:c8:29:90:2d:9c:9f:25:ca:7a)
-	key := "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAAAYQDThS/Tvat+hiKTsZ0fuaSkd/o8cMMufy70BXE2/UPZhUx/M4/XBGGiHGXwEbxcRpSTmEsAyN5vGt7nnZrHsjq/WhZmCjVcNlyfnEMGD0UgfsjwVDTT3Y4BYN62bhW4LfM="
+	// ssh public key for testing (fingerprint: 16:c4:b0:2b:99:b4:eb:6d:06:a3:7d:7f:d6:55:e1:65)
+	key := "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDZ4khAJykHnJXKbv2mWgegSVyVKG6mAl6MwlMjGP9CsuBsbgGrMxCVVH1lhRIWNyGNPh3tqt5qmz0rSjaRqBO9FbTxTXbokAbktWgLvSH3PqSKXyS8NmsaMk/PfAm9BOwYpVQ6P2dg75MmKbXgq1xxe3PxEmgdEHzg4XSw+dkoO4Bika88BksFaXHsE1DIlqfDNtTIheq6vccJ2L9bEoXdhsEbe/OaZsyZHeaOg2Z7Z3K531kLfAYH/uD3kUOcA+4sX/tW29aYpVvca7v1sHFjz9RotF3PkxRH/zekLy8/WHxfxwOptBd0TpEiI0Vg6Yy+pNjkSEGwKNPO8Z2aEflT"
```
Comment (parkr) [inline tests/integration/users_test.go]: Why was this changed?

**Comment by parkr** [comment]
@bjoernhaeuser, @willnorris handles review for this repo. He may need a direct ping, depending upon how he watches the repo's issues and pull requests.

File: tests/integration/users_test.go (hunk 2 above)
Comment (bjoernhaeuser) [inline tests/integration/users_test.go]: The old one was too short, github requires a minimum of 2048 bits for the key.

File: README.md (hunk 1 above)
Comment (bjoernhaeuser) [inline README.md]: Absolutly, should that be removed?

Just wanted to represent whats the new minimum according to that commit here: https://github.com/google/go-github/commit/92bbb1ce51b9d7dc10debfd354d84fc8c2b287f5

**Comment by willnorris** [comment]
nothing much in particular is holding up reviewing this.. I just got a little busy.  

I'll probably end up regenerating a new key for the user integration test myself just because of the security implications involved (see comment on 19c0916d4b04896a0002bd73d4a341834ea1e82a), but otherwise this is *probably* all fine... I'll just need to take a closer look in a bit.

**Comment by willnorris** [comment]
okay, I split this up and merged as 68d9199c291c71452d245b906823462551185acd and cedb0a9ae0c3843037194211a692c503ab83c33e.  New key was added in b8b4ac742977310ff6e75140a403a38dab109977.  Thanks @bjoernhaeuser!
